### PR TITLE
Add layout styles and templates

### PR DESCRIPTION
**Pull request by sbrack8t**

### CODE REVIEW

File: _starter-kit/_patterns/02-layouts/flex/_flex.scss [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+.l-flex-media {
+  display: flex;
+  
+  .l-flex-media__rich-media {
+    flex: none;
+    margin: 0 $flex-gutter 0 0;
+  }
+
+  .l-media-object__content {
```
Comment (coreylafferty) [inline _starter-kit/_patterns/02-layouts/flex/_flex.scss]: Shouldn't this be .l-flex-media__content?

File: _starter-kit/_patterns/02-layouts/flex/_flex.scss [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    margin: 0 0 0 $flex-gutter;
+    order: 1;
+  }
+}
+
+.l-flex-media--column { 
+  flex-direction: column;
+}
+
+.l-flex-row {
```
Comment (coreylafferty) [inline _starter-kit/_patterns/02-layouts/flex/_flex.scss]: Can the styles in this file be separated into _l-flex-row.scss and _l-flex-media.scss?

File: _starter-kit/_patterns/02-layouts/flex/flex-media.twig [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<div class="l-flex-media {{ flex_modifier_classes }}">
+  <div class="l-flex-media__media">{{ flex_media }}</div>
+  <div class="l-flex-media__content">{{ flex_content }}</div>
+</div>Lorem, ipsum dolor sit amet consectetur adipisicing elit. Molestiae necessitatibus fuga provident aut.
```
Comment (coreylafferty) [inline _starter-kit/_patterns/02-layouts/flex/flex-media.twig]: Should this lorem ipsum text be here?

File: _starter-kit/_patterns/02-layouts/grids/_grid.scss [hunk 4]
```diff
@@ -0,0 +1,92 @@
+// @file
+// Styles for the grid layout.
+$bp-grid-small: 700px;
+$bp-grid-large: 1000px;
+$bp-grid-mega: 1300px;
+
+$grid-item-minwidth: 400px;
+$grid-gutter: $layout-gutter-width;
+
+.l-grid-3col {
```
Comment (coreylafferty) [inline _starter-kit/_patterns/02-layouts/grids/_grid.scss]: should these styles be reorganized to use variants (e.g., l-grid, l-grid--3col, l-grid--4col)?

Comment (sbrack8t) [reply _starter-kit/_patterns/02-layouts/grids/_grid.scss]: I'll consolidate the initial grid set up with a placeholder. But I'm not sure who much is gained by further consolidation and variant overrides. I can see the potential for a l-grid-[#]col--[modifier] variant where you modify a single or group of cells in the grid. But I think we should leave the column # as separate entities.

**Pull request merged**